src/test/java/run/halo/app/controller/AdminControllerTest.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @Test
+    public void resetPasswordShouldOk() throws Exception {
+        ResetPasswordParam param = new ResetPasswordParam();
+        param.setUsername("admin");
+        param.setEmail("example@example.com");
+        param.setPassword("a password");
+        param.setCode("a code");
+        param.setPassword("12345678");
+        resetPasswordPerform(JsonUtils.objectToJson(param))
+            .andDo(print())
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void resetPasswordAndEmailFieldAbsentShouldReturn4xxError() throws Exception {
+        ResetPasswordParam param = new ResetPasswordParam();
+        // The email field value in the parent class is missing，verification will also be triggered
+        param.setUsername("admin");
+        param.setCode("a code");
+        param.setPassword("a password");
+        resetPasswordPerform(JsonUtils.objectToJson(param))
+            .andDo(print())
+            .andExpect(status().is4xxClientError())
+            .andExpect(jsonPath("$.message").value(FIELD_ERROR_MESSAGE));
+    }
+
+    @Test
+    public void resetPasswordAndCodeFieldAbsentShouldReturn4xxError() throws Exception {
+        ResetPasswordParam param = new ResetPasswordParam();
+        // The code field value in the param class is missing，verification will also be triggered
+        param.setUsername("admin");
+        param.setEmail("example@example.com");
+        param.setPassword("a password");
+        resetPasswordPerform(JsonUtils.objectToJson(param))
+            .andDo(print())
+            .andExpect(status().is4xxClientError())
+            .andExpect(jsonPath("$.message").value(FIELD_ERROR_MESSAGE));
+    }
+
+    @Test
+    public void resetPasswordAndInsufficientPasswordLengthShouldReturn4xxError() throws Exception {
+        ResetPasswordParam param = new ResetPasswordParam();
+        // The code field value in the param class is missing，verification will also be triggered
+        param.setUsername("admin");
+        param.setEmail("example@example.com");
+        param.setCode("a code");
+        param.setPassword("123");
+        resetPasswordPerform(JsonUtils.objectToJson(param))
+            .andDo(print())
+            .andExpect(status().is4xxClientError())
+            .andExpect(jsonPath("$.message").value(FIELD_ERROR_MESSAGE));
+    }
+
+    @NotNull
+    private ResultActions resetPasswordPerform(String param)
+        throws Exception {
+        return this.mockMvc.perform(put("/api/admin/password/reset")
+            .content(param)
+            .contentType(MediaType.APPLICATION_JSON));
+    }
+
+    @NotNull
+    private ResultActions sendResetPasswordCodePerform(String param)
+        throws Exception {
+        return this.mockMvc.perform(post("/api/admin/password/code")
+            .content(param)
+            .contentType(MediaType.APPLICATION_JSON));
+    }
+
+    public static class MockAdminService implements AdminService {
```

Comment:
No need to create a blank AdminService.

Reply:
Please be aware of that this is a test for AdminController, not only for parameter verification.

Reply:
This issue won't block this PR from merging. Anyway, we are supposed to consider optimizing unit-test experience later.